File: calc/python/sol/apigear_calc/calculator.py
from apigear_calc.api import api

class Calculator(api.ICalculator):
    def __init__(self):
        self.__total : int = 0

    @property
    def total(self):
        return self.__total

    def add(self, a: int) -> None:
        self.__total += a

    def substract(self, a: int) -> None:
        self.__total -= a

    def multiply(self, a: int) -> None:
        self.__total *= a

    def divide(self, a: int) -> None:
        self.__total /= a

    def clear(self) -> None:
        self.__total = 0

File: calc/typescript/sol/src/apigear_calc/calculator.ts
import * as api from "./api";

export class Calculator implements api.ICalculator {
  total = 0;
  add(a: number): void {
    this.total += a;
  }
  substract(a: number): void {
    this.total -= a;
  }
  multiply(a: number): void {
    this.total *= a;
  }
  divide(a: number): void {
    this.total /= a;
  }
  clear(): void {
    this.total = 0;
  }
}
